ui/src/app/cohort-search/demographics/demographics.component.tsx

```diff
@@ -0,0 +1,592 @@
+const ageNode = {
+  hasAncestorData: false,
+  attributes: [],
+  code: '',
+  domainId: DomainType.PERSON,
+  group: false,
+  name: 'Age',
+  parameterId: 'age-param',
+  isStandard: true,
+  type: CriteriaType.AGE,
+  value: ''
+};
+
+const ageTypes = [
+  {label: 'Current Age', type: AttrName.AGE.toString()},
+  {label: 'Age at Consent', type: AttrName.AGEATCONSENT.toString()},
+  {label: 'Age at CDR Date', type: AttrName.AGEATCDR.toString()}
+];
+
+const defaultMinAge = '18';
+const defaultMaxAge = '120';
+
+function sortByCountThenName(critA, critB) {
+  const A = critA.count || 0;
+  const B = critB.count || 0;
+  const diff = B - A;
+  return diff === 0
+        ? (critA.name > critB.name ? 1 : -1)
+        : diff;
+}
+interface Props {
+  select: Function;
+  selectedIds: Array<string>;
+  selections: Array<any>;
+  wizard: any;
+}
+
+interface State {
+  ageType: any;
+  ageTypeNodes: any;
+  calculating: boolean;
+  count: number;
+  loading: boolean;
+  maxAge: string;
+  minAge: string;
+  nodes: Array<any>;
+}
+
+export class Demographics extends React.Component<Props, State> {
+  ageWrapper: HTMLDivElement;
+  slider: any;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      ageType: AttrName.AGE.toString(),
+      ageTypeNodes: undefined,
+      calculating: false,
+      count: null,
+      loading: true,
+      maxAge: defaultMaxAge,
+      minAge: defaultMinAge,
+      nodes: undefined,
+    };
+  }
+
+  componentDidMount(): void {
+    if (this.props.wizard.type === CriteriaType.AGE) {
+      if (serverConfigStore.getValue().enableCBAgeTypeOptions) {
+        this.loadAgeNodesFromApi();
+      } else {
+        this.setState({loading: false});
+      }
+    } else {
+      this.loadNodesFromApi();
+    }
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>): void {
+    const {selections, wizard} = this.props;
+    if (selections !== prevProps.selections && wizard.type !== CriteriaType.AGE) {
+      this.calculate();
+    }
+  }
+
+  loadNodesFromApi() {
+    const {selections, wizard} = this.props;
+    const {cdrVersionId} = currentWorkspaceStore.getValue();
+    this.setState({loading: true});
+    cohortBuilderApi().findCriteriaBy(+cdrVersionId, DomainType.PERSON.toString(), wizard.type).then(response => {
+      const nodes = response.items
+        .filter(item => item.parentId !== 0)
+        .sort(sortByCountThenName)
+        .map(node => ({...node, parameterId: `param${node.conceptId || node.code}`}));
+      if (selections.length) {
+        this.calculate(true);
+      }
+      this.setState({loading: false, nodes});
+    });
+  }
+
+  loadAgeNodesFromApi() {
+    const {cdrVersionId} = currentWorkspaceStore.getValue();
+    const initialValue = {
+      [AttrName.AGE.toString()]: [],
+      [AttrName.AGEATCONSENT.toString()]: [],
+      [AttrName.AGEATCDR.toString()]: []
+    };
+    cohortBuilderApi().findAgeTypeCounts(+cdrVersionId).then(response => {
```

Comment:
We have a code style preference to use `await` here, rather than `.then`. I have no clue if this has been broadcast to the y'all? (I think we talked about it at the eng sync?)

So this could be done as:
```
const ageTypeNodes = await cohortBuilderApi().findAgeTypeCounts(+cdrVersionId).items.reduce((acc, item) => {
acc[item.ageType].push(item);
return acc;
}, initialValue);
this.setState({ageTypeNodes}, () => this.calculateAgeFromNodes());

Reply:
Ok, good to know. Is there style guide somewhere? Seems like I've heard it mentioned but don't think I've seen the doc before.

Reply:
Yes! The style guide is here: https://github.com/all-of-us/workbench/blob/master/ui/docs/style-guide.md

Reply:
Perfect, thanks!